src/docs/build-deploy-and-maintain-apps/build-an-application.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Build an Application
+
+slug: build-an-application
+
+description: <TBD>
+
+keywords: build, application, openshift, hardening, security, SAST, DAST, PaaS, Risk Mitigation, 
+
+page_purpose: 
+
+audience: technical lead, developer
+
+author: Kevin Summersill
+
+content_owner: 
+
+sort_order: 
+---
+# Build an Application
+This documenation identify the following:
+*   Best Practices when working with Images
+*   Security Best Practices to utilize Before, During, and After pushing to production. 
+
+- [Build an Application](#build-an-application)
+  * [Minimizing Server Image Surface Area](#minimizing-server-image-surface-area)
+    + [What are bloated images?](#what-are-bloated-images-)
+    + [What are the reasons to reduce image bloat?](#what-are-the-reasons-to-reduce-image-bloat-)
+    + [Best Practices for Reducing Image Bloat:](#best-practices-for-reducing-image-bloat-)
+  * [Shifting Left and Mitigating Risk](#shifting-left-and-mitigating-risk)
+    + [So what is .devcontainer?](#so-what-is-devcontainer-)
+    + [Best Practices for .devcontainer:](#best-practices-for-devcontainer-)
+  * [Testing the Server-Based Images](#testing-the-server-based-images)
+  * [Securing and Hardening Images](#securing-and-hardening-images)
+  * [Integrating Security in the Pipelines with Best Practices](#integrating-security-in-the-pipelines-with-best-practices)
+  * [Utilizing Semantic Versioning with Images](#utilizing-semantic-versioning-with-images)
+
+## Minimizing Server Image Surface Area ## 
+There are multiple benefits of trimming the fat from images before deploying them into the Platform as a Service. This section of the documentation will cover these benefits:
+
+### What are bloated images? ###
+A bloated image is an image that contains increased amounts of unnecessary items which are not needed to run the application. Examples of unnecessary items are packages, binaries, and libraries that may be installed or on the image that are not being used. 
+### What are the reasons to reduce image bloat? ###
+*   Packages, Libraries, and binaries that are not tracked or managed within the image can cause security risks. 
+*   The increased image size can cause the start time of images to increase when immutable. For example, when images are immutable and set up in a stateless manner they may be rescheduled to nodes. If the image pull policy is set up to “If Present” or set up to “Always” then the image will need to be reloaded. This could potentially increase the amount of time that it takes for the service to be available. 
+### Best Practices for Reducing Image Bloat: ###
+*   Utilizing Universal Binary Images or Minimal Based Images 
+*   When building utilizing Multi-Stage Images and removing building tools. (e.g. Java Development Kit)
```

Comment:
Maybe include a link to docs on how to do this?
https://docs.openshift.com/container-platform/4.10/cicd/builds/advanced-build-operations.html#builds-chaining-builds_advanced-build-operations
https://docs.docker.com/build/building/multi-stage/